segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/WriteOperation.java

```diff
@@ -452,4 +496,48 @@ private CompletableFuture<Void> writeToChunk(MetadataTransaction txn,
                     return v;
                 }, chunkedSegmentStorage.getExecutor());
     }
+
+    private CompletableFuture<Void> validateWrittenData(ChunkHandle chunkHandle, long startOffsetInChunk, int startOffestInInputData, int bytesCount) {
+        val bufferForRead = new byte[bytesCount];
+        return readChunk(chunkHandle, startOffsetInChunk, bytesCount, bufferForRead)
+                .thenAcceptAsync(v -> {
+                    val mismatch = Arrays.mismatch(bufferForRead, 0, bytesCount,
+                            inputData.get(), startOffestInInputData, startOffestInInputData + bytesCount);
+                        Preconditions.checkState(-1 == mismatch, "Data read from chunk differs from data written at offset %s",
```

Comment:
Same here: throwing a `DataCorruptionException` may help us to detect when this problem occurs.

Reply:
DataCorruptionException is defined in io.pravega.segmentstore.server. Storage can not depend on that project.